lib/manager/composer/artifacts.js

```diff
@@ -38,17 +38,24 @@ async function getArtifacts(
     if (!config.gitFs) {
       await fs.outputFile(localLockFileName, existingLockFileContent);
     }
-    let authJson;
-    const credentials = hostRules.find({
+    let authJson = {};
+    let credentials = hostRules.find({
       platform: 'github',
       host: 'api.github.com',
     });
     // istanbul ignore if
     if (credentials && credentials.token) {
-      authJson = {
-        'github-oauth': {
-          'github.com': credentials.token,
-        },
+      authJson['github-oauth'] = {
+        'github.com': credentials.token,
+      };
+    }
+    credentials = hostRules.find({
+      platform: 'gitlab',
+      host: 'gitlab.com',
+    });
+    if (credentials && credentials.token) {
+      authJson['gitlab-oauth'] = {
```

Comment:
GitLab treats personal access tokens and oauth tokens completely differently, so this may not work. It’s there an option for a non-oauth GitLab token in auth.json?

Reply:
Theres `gitlab-oauth` and `gitlab-token`. Which one should this be using?

https://getcomposer.org/doc/06-config.md#gitlab-oauth